security/custom-gas-token.md

```diff
@@ -0,0 +1,83 @@
+
+### Non Standard ERC20 Usage
+
+- **Description:**
+    - This is related to [Bridge Undercollateralization](https://www.notion.so/Bridge-Undercollateralization-552a985fcbed4d95b468f4b4c5f22ec9?pvs=21) as it is one form that the bridge can become undercollaterized. Really anything is possible with a non standard ERC20
+- **Risk Assessment:**
+    - High severity, low likelihood
+    - Given that we communicate with rollup as a service platforms, this is low probability. Chains that don’t listen to us can do whatever they want in their world but they will never be able to be considered standard.
+- **Mitigations:**
+    - We should not co-market with any custom gas token chain unless we have manually checked that their ERC20 token complies with the specs.
+- **Detection:** *How do we detect if this occurs?*
+    - Automation cannot fully detect that the ERC20 token meets the standardness checks.
+- **Recovery Path(s)**:
+    - We will need to remove the chain from the superchain if they are able to launch without us realizing that they are not meeting the ERC20 standardness requirements. It is risky to have the team migrate to a standard ERC20 because it would require a full solvency check first.
+
+### Withdrawal Bricked Bug
+
+- **Description:** *Details of the failure mode go here. What the causes and effects of this failure?*
+    - It is possible that a bug exists that would prevent users to withdraw that is unrelated to bridge undercollaterization. This sort of bug would just prevent users from withdrawing.
+- **Risk Assessment:** *Simple low/medium/high rating of impact (severity) + likelihood.*
+    - High severity, low likelihood
+    - There is test coverage for this, both unit testing and end to end testing. This has also been tested on live networks.
+- **Mitigations:** *What mitigations are in place, or what should we add, to reduce the chance of this occurring?*
+    - The live testing has shown that networks can operate with custom gas token enabled, see [https://github.com/ethereum-optimism/specs/discussions/140#discussioncomment-9426636](https://github.com/ethereum-optimism/specs/discussions/140#discussioncomment-9426636)
+- **Detection:**
+    - This isn’t immediately obvious if there is an automated way to detect but users will certainly let us know that there is a bug.
+- **Recovery Path(s)**:
+    - This may require an irregular state transition to fix
+
+## Action Items
+
+Below is what needs to be done before launch to reduce the chances of the above failure modes occurring, and to ensure they can be detected and recovered from:
+
+- [ ]  Figure out existing chain monitoring solutions and if rollup as a service providers operate them
+    - [ ]  If yes, then design the necessary changes to the monitoring service so that they can eventually be implemented
+- [ ]  *Action item 3 (Assignee: tag assignee)*
```

Comment:
We need to ensure that our docs reflect the risks of using non-standard ERC20 tokens.